app/models/services.py

```diff
@@ -12,67 +15,123 @@ class ErrInvalidParameters(Exception):
     pass
 
 
-# DB initializtion
-db = tinydb.TinyDB("db.json")
-services = db.table("services")
-
-
-def insert(shop_id="", name="", description="", images=None):
-    if not images:
-        images = []
-
-    # Initialize id with randomly generated string
-    service_id = util.random_id(initial="s")
-
-    services.insert(
-        {
-            "service_id": service_id,
-            "shop_id": shop_id,
-            "name": name,
-            "slug": slugify(name),
-            "description": description,
-            "images": images,
-        }
-    )
-
-    # TODO: upload images into disk
-    pass
-
-    return service_id
-
-
-def delete(service_id):
-    query = tinydb.Query()
-    services.remove(query.service_id == service_id)
-
-
-def all():
-    results = services.all()
-
-    if not results:
-        raise ErrNotFound("oops.. no service found in the db")
-
-    return results
-
-
-def get(service_id):
-    service = tinydb.Query()
-    result = services.search(service.service_id == service_id)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided id: {service_id}")
-
-    return result[0]
-
-
-def get_by_name(name=""):
-    if not name:
-        raise ErrInvalidParameters("service name must be provided")
-
-    service = tinydb.Query()
-    result = services.search(service.name == name)
-
-    if not result:
-        raise ErrNotFound(f"no service found for the provided name: {name}")
-
-    return result[0]
+class ServiceRepo:
+
+    def __init__(self, db: tinydb.TinyDB):
+        self.db = db
+        self.services = db.table('services')
+
+    def insert(self, id="",
+               shop_id="",
+               name="",
+               slug="",
```

Comment:
I think no need to pass slug as a parameter, we can just generate it on the go inside the insert method

Reply:
I'll update it , thanks